instr.ml

```diff
@@ -322,3 +322,82 @@ let checkpoint_label pc =
 
 let independent instr exp =
   VarSet.is_empty (VarSet.inter (changed_vars instr) (expr_vars exp))
+
+class mapper = object (m)
+  method variable x = x
```

Comment:
is this one really necessary. found it very confusing. just inline below. i think users should think about the difference. i don't see a reasonable justification for unifying the handling of the *rhs* of an assignment and a *variable* occuring in an expression (except that they happen to have the same type in our instrs).